### PR TITLE
Bulk import students

### PR DESCRIPTION
**Pull request by karenc**
- Update openstax_accounts to 5.2.0
- Change edit_course_spec.rb to a feature test
- Add an admin form for creating a period for a course
- Add an admin form to edit periods
- Style flash errors in admin screens
- Change UserProfile::GetAccount to find by username or id
- Change UserProfile::CreateProfile to use email and/or username
- Add more optional arguments for UserProfile::CreateProfile
- Add admin page for uploading student roster
- Fix GetStudentRoster spec by sorting the result

### CODE REVIEW

**Comment by Dantemss** [comment]
Not sure if this will help, but you could look at how Nathan imports excel spreadsheets in https://github.com/openstax/exercises/blob/master/app/routines/exercises/import/excel.rb

**Comment by karenc** [comment]
@Dantemss Thanks! We're using CSV files.

File: app/views/admin/courses/_students.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<% method ||= :post %>
+
+<%= lev_form_for(:course,  url: url, method: method, html: { multipart: true }) do |f| %>
```
Comment (jpslav) [inline app/views/admin/courses/_students.html.erb]: don't think you need to use `lev_form_for` unless you're using handlers (which you could do for the `CoursesController#students` method (but don't have to)

File: app/controllers/admin/courses_controller.rb [hunk 2]
```diff
@@ -36,4 +60,22 @@ def course_params
...
+      AddUserAsPeriodStudent.call(period: period, user: profile.entity_user)
+    end
+  end
+
+  def create_user(user)
+    profile = UserProfile::CreateProfile[email: user['email'], username: user['username']]
+    first_name = user['first_name']
+    last_name = user['last_name']
+    full_name = "#{first_name} #{last_name}" if first_name.present? && last_name.present?
+    profile.account.update_attributes(first_name: first_name,
```
Comment (jpslav) [inline app/controllers/admin/courses_controller.rb]: Are these updates propagated to OpenStax Accounts?

File: config/routes.rb [hunk 3]
```diff
@@ -80,7 +80,12 @@
 
     resources :administrators, only: [:index, :create, :destroy]
 
-    resources :courses, except: :destroy
+    resources :courses, except: :destroy do
+      member do
+        post 'students'
+      end
+      resources :periods, except: :destroy
```
Comment (jpslav) [inline config/routes.rb]: Might ask you to do periods destroy in a different PR.

**Comment by jpslav** [comment]
Looks good @karenc just one question in review

File: app/controllers/admin/courses_controller.rb (hunk 2 above)
Comment (karenc) [inline app/controllers/admin/courses_controller.rb]: They're definitely updated in the openstax_accounts_accounts table.  I tried using a local accounts instance but I don't see any users created in the accounts database.

Comment (karenc) [inline app/controllers/admin/courses_controller.rb]: Actually I was checking the wrong database, the users are there, but the names are not getting updated.

Comment (jpslav) [inline app/controllers/admin/courses_controller.rb]: @Dantemss what's the mechanism for getting these local updates to Accounts?

**Comment by karenc** [comment]
Not sure why but now that I'm connected to an accounts instance, I'm getting this error:

```
ArgumentError in Admin::CoursesController#students
You tried to define an enum named "role_type" on the model "Entity::Role", but this will generate a instance method "role_type=", which is already defined by another enum.

app/models/entity/role.rb:2:in `<class:Role>'
app/models/entity/role.rb:1:in `<top (required)>'
app/subsystems/role/create_user_role.rb:1:in `require_relative'
app/subsystems/role/create_user_role.rb:1:in `<top (required)>'
app/routines/add_user_as_period_student.rb:6:in `<class:AddUserAsPeriodStudent>'
app/routines/add_user_as_period_student.rb:1:in `<top (required)>'
app/controllers/admin/courses_controller.rb:67:in `block in add_students'
app/controllers/admin/courses_controller.rb:65:in `each'
app/controllers/admin/courses_controller.rb:65:in `add_students'
app/controllers/admin/courses_controller.rb:48:in `students'
```

**Comment by karenc** [comment]
The problem with Entity::Role enum "role_type" can be fixed by either restarting the server or add cache_classes = true in the development.rb (@Dantemss)

File: app/controllers/admin/courses_controller.rb (hunk 2 above)
Comment (karenc) [inline app/controllers/admin/courses_controller.rb]: It seems it is triggering an update to accounts, but I get this error:

```
OAuth2::Error in Admin::CoursesController#students

 SecurityTransgression in Api::V1::UsersController#update

SecurityTransgression

Rails.root: /home/karen/accounts
Application Trace | Framework Trace | Full Trace

app/controllers/api/v1/users_controller.rb:119:in `update' lib/omniauth/strategies/custom_identity.rb:49:in `other_phase'
```

Comment (karenc) [inline app/controllers/admin/courses_controller.rb]: It seems accounts doesn't allow the names of temp users.  So updating the name locally is the way to go for now.
This PR will make updating names easier: https://github.com/openstax/accounts-rails/pull/29

Comment (jpslav) [inline app/controllers/admin/courses_controller.rb]: merged that PR

**Comment by karenc** [comment]
I've made the following changes:

- Updated openstax_accounts to 5.2.0
- Updated `UserProfile::CreateProfile` optional arguments to match `OpenStax::Accounts::CreateTempAccount`
- Use `OpenStax::Accounts::CreateTempAccount` to update the name columns

Ready for review.
